packages/next-auth/src/index.tsx

```diff
@@ -233,7 +233,7 @@ export interface NextAuthResult {
     ((...args: []) => Promise<Session | null>) &
     ((...args: [GetServerSidePropsContext]) => Promise<Session | null>) &
     ((
-      ...args: [(req: NextAuthRequest) => ReturnType<AppRouteHandlerFn>]
+      ...args: [(req: NextAuthRequest, ctx: AppRouteHandlerFnContext) => ReturnType<AppRouteHandlerFn>]
```

Comment:
```suggestion
      ...args: [(..._args: Parameters<AppRouteHandlerFn>) => ReturnType<AppRouteHandlerFn>]
```
Then no need to export the `AppRouteHandlerFnContext` type

Reply:
That won't work, because the AppRouteHandlerFn takes a `NextRequest`, not a `NextAuthRequest`, so the `.auth` param won't be available:
https://github.com/nextauthjs/next-auth/blob/5c3cc2e9facd0ace1da4aac64cacd8c726a837ee/packages/next-auth/src/lib/types.ts#L14

Reply:
I see, then this is fine, resolving.